signer/src/storage/postgres.rs

```diff
@@ -2121,6 +2125,46 @@ impl super::DbRead for PgStore {
         .map_err(Error::SqlxQuery)
     }
 
+    async fn is_withdrawal_active(
+        &self,
+        id: &model::QualifiedRequestId,
+        bitcoin_chain_tip: &model::BitcoinBlockRef,
+        min_confirmations: u64,
+    ) -> Result<bool, Error> {
+        let min_block_height = sqlx::query_scalar::<_, Option<i64>>(
+            r#"
+            SELECT MAX(bb.block_height) + 1
+            FROM sbtc_signer.bitcoin_withdrawals_outputs AS bwo
+            JOIN sbtc_signer.bitcoin_blocks AS bb
+              ON bb.block_hash = bwo.bitcoin_chain_tip
+            WHERE bwo.request_id = $1
+              AND bwo.stacks_block_hash = $2
+            "#,
+        )
+        .bind(i64::try_from(id.request_id).map_err(Error::ConversionDatabaseInt)?)
+        .bind(id.block_hash)
+        .fetch_one(&self.0)
+        .await
+        .map_err(Error::SqlxQuery)?;
+
+        // This means that there are no rows associated with the ID in the
+        // `bitcoin_withdrawals_outputs` table.
+        let Some(min_block_height) = min_block_height else {
+            return Ok(false);
+        };
+
+        let output_type = model::TxOutputType::SignersOutput;
+        let chain_tip_hash = &bitcoin_chain_tip.block_hash;
+        let signer_utxo_fut = self.get_utxo(chain_tip_hash, output_type, min_block_height);
+        // If this returns None, then the sweep itself could be in the
+        // mempool. If that's the case then this is definitely active.
+        let Some(signer_utxo) = signer_utxo_fut.await? else {
+            return Ok(true);
+        };
+
+        Ok(signer_utxo.block_height + min_confirmations >= bitcoin_chain_tip.block_height)
```

Comment:
mmm `get_utxo` always return the latest utxo, so if we have a sweep every couple of blocks every withdrawal will always be active, as even for one way in the past we would still get a recent utxo (and `bitcoin_chain_tip.block_height` will always be less than `min_confirmations` ahead in the example above).

Reply:
Bah! What was I thinking, of course it returns an unspent transaction output 🤦🏽‍♂️. But an easy fix.